sharding/node/backend.go

```diff
@@ -148,53 +149,50 @@ func (s *ShardEthereum) Close() {
 	close(s.stop)
 }
 
-// Register appends a service constructor function to the service registry of the
+// registerService appends a service constructor function to the service registry of the
 // sharding node.
-func (s *ShardEthereum) Register(constructor sharding.ServiceConstructor) error {
-	s.lock.Lock()
-	defer s.lock.Unlock()
-
-	ctx := &sharding.ServiceContext{
-		Services: make(map[reflect.Type]sharding.Service),
-	}
-
-	// Copy needed for threaded access.
-	for kind, s := range s.services {
-		ctx.Services[kind] = s
-	}
-
-	service, err := constructor(ctx)
-	if err != nil {
-		return err
-	}
-
+func (s *ShardEthereum) registerService(service sharding.Service) error {
 	kind := reflect.TypeOf(service)
 	if _, exists := s.services[kind]; exists {
 		return fmt.Errorf("service already exists: %v", kind)
 	}
 	s.services[kind] = service
-
+	s.serviceTypes = append(s.serviceTypes, kind)
 	return nil
 }
 
+// fetchService returns a reference to a service stored in the service registry.
```

Comment:
The comments should mention that this returns a copy of the original service, since its an important gotcha. This could also return a pointer to the same service but we would have to be mindful of concurrency issues.